extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -1189,6 +1190,38 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 			return false;
 		}
 	}
+
+	//Get QueryProvider
+	private async runQueryWithResults(queryString: string): Promise<azdata.SimpleExecuteResult> {
+		const connectionProfile = await this.getSourceConnectionProfile();
+		const connectionUri = await azdata.connection.getUriForConnection(this._sourceConnectionId);
+		const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>(connectionProfile.providerId, azdata.DataProviderType.QueryProvider);
+
+		return await queryProvider.runQueryAndReturn(connectionUri, queryString);
+	}
+
+	public async getDatabasesList(): Promise<azdata.DatabaseInfo[]> {
+		try {
+			const queryResult = await this.runQueryWithResults(queries.SELECT_ALL_DB_W_SIZE);
+
+			const result = queryResult.rows.map(row => {
+				return {
+					options: {
+						name: getSqlString(row[0]),
+						state: getSqlString(row[1]),
+						sizeInMB: getSqlString(row[2]),
+					}
+				};
+			}) ?? [];
+
+			return result;
+		} catch (error) {
+			console.error(error);
```

Comment:
nit: `logError` below already writes to console so this is probably unnecessary

Reply:
Removed.